### PR TITLE
Readme fixes

### PR DESCRIPTION
**Pull request by kaechele**
Here are some fixes for the README.
The changes should help clarify that Istio injection is required for APIClarity to work.

I also edited some of the wording and removed the redundant first paragraph, which was almost identical to the "Solution" paragraph.

This file now also successfully validates against [markdownlint-cli](https://github.com/igorshubovych/markdownlint-cli).

If you have any feedback let me know or feel free to edit this PR directly.

### CODE REVIEW

**Comment by FrimIdan** [comment]
@kaechele tnx for the fixes.

> This file now also successfully validates against markdownlint-cli.

Do you think we should add `markdownlint-cli` run in the PR GitHub action?

File: README.md [hunk 1]
```diff
@@ -1,105 +1,190 @@
...
+   kubectl label namespaces sock-shop istio-injection=enabled
+   ```
+
+3. From the `microservices-demo` git repository cloned earlier deploy the demo
+   to your cluster:
+
+   ```shell
+   cd microservices-demo
+   kubectl apply -f deploy/kubernetes/complete-demo.yaml
+   ```
```
Comment (FrimIdan) [inline README.md:136]: We can just do
1. Create the `sock-shop` ...
2. Deploy the demo... `kubectl apply -f https://raw.githubusercontent.com/microservices-demo/microservices-demo/master/deploy/kubernetes/complete-demo.yaml`

WDYT?

Comment (kaechele) [reply README.md]: Great idea! Added the suggestion and rebased onto master.

**Comment by kaechele** [comment]
> @kaechele tnx for the fixes.
> 
> > This file now also successfully validates against markdownlint-cli.
> 
> Do you think we should add `markdownlint-cli` run in the PR GitHub action?

Yes, that's an excellent idea.

Once this lands I'll create a PR to add a [pre-commit](https://pre-commit.com/) config so that developers can op to use pre-commit to catch stuff that would otherwise be flagged in the linting stage of the GitHub actions. Should improve code and documentation quality even before it's committed. It's completely optional and up to the individual developer to enable though.

**Comment by kaechele** [comment]
Also, if you like I can squash these changes into one commit before pulling.

**Comment by FrimIdan** [comment]
> Also, if you like I can squash these changes into one commit before pulling.

sure please squash

**Review by FrimIdan — APPROVED** [review]

**Comment by kaechele** [comment]
Squashed and force pushed.

**Pull request merged**